rest_flex_fields/serializers.py

```diff
@@ -17,33 +17,41 @@ class FlexFieldsSerializerMixin(object):
     def __init__(self, *args, **kwargs):
         self.expanded_fields = []
 
-        passed = { 
-            'expand' : kwargs.pop('expand', None), 
-            'fields': kwargs.pop('fields', None) 
+        passed = {
+            'expand': kwargs.pop('expand', None),
+            'fields': kwargs.pop('fields', None),
+            'omit': kwargs.pop('omit', [])
         }
 
+        # add excludes from expandable_fields to those on query params
+        passed['omit'] += kwargs.pop('exclude', [])
+
         super(FlexFieldsSerializerMixin, self).__init__(*args, **kwargs)
         expand = self._get_expand_input(passed)
         fields = self._get_fields_input(passed)
+        omit = self._get_omit_input(passed)
         expand_field_names, next_expand_field_names = split_levels(expand)
         sparse_field_names, next_sparse_field_names = split_levels(fields)
-        expandable_fields_names = self._get_expandable_names(sparse_field_names)
+        omit_field_names, next_omit_field_names = split_levels(omit)
+        omit_field_names = set(omit_field_names) - next_omit_field_names.keys()
```

Comment:
what is the purpose of this line?

Reply:
This is because we don't want to remove any parent fields when omitting something from the child, e.g. omit=foo,nested.bar, omit_field_names will contain ['foo', 'nested'] and next_omit_field_names {'nested': ['bar']}; we only want to omit 'foo' on this pass so remove nested since it is in the keys of the next set